xcom2-launcher/xcom2-launcher/Forms/CleanModsForm.cs

```diff
@@ -24,35 +24,47 @@ public CleanModsForm(Settings settings)
 
         private void onStartButtonClicked(object sender, EventArgs e)
         {
-            var result = MessageBox.Show("Are you sure?\r\nThis might cause problems and can not be undone.", "Confirm", MessageBoxButtons.OKCancel);
-            if (result != DialogResult.OK)
+            var dialogResult = MessageBox.Show("Are you sure?\r\nThis might cause problems and can not be undone.", "Confirm", MessageBoxButtons.OKCancel);
+            
+            if (dialogResult != DialogResult.OK)
                 return;
             
             var source_mode = source_groupbox.Controls.OfType<RadioButton>().FirstOrDefault(r => r.Checked).Name;
             var shader_mode = shader_groupbox.Controls.OfType<RadioButton>().FirstOrDefault(r => r.Checked).Name;
 
             foreach (var m in Mods.All)
             {
-                // Source Files
-                if (hasSourceFiles(m) && !hasCookedPCConsole(m))
+                try
                 {
-                    if (source_mode == "src_all_radiobutton")
-                        deleteSourceFiles(m);
+                    // Source Files
+                    if (hasSourceFiles(m) && !hasCookedPCConsole(m))
+                    {
+                        if (source_mode == "src_all_radiobutton")
+                            deleteSourceFiles(m);
+
+                        else if (source_mode == "src_xcomgame_radiobutton")
+                            if (hasXComGameSourceFiles(m))
```

Comment:
Oh god. Braces please

Reply:
You mean the fact that the original author left out braces for the one liners? I do not like that as well, but preferred not to touch anything that wasn't needed. 
I have no clue what the cleaning stuff is supposed to do in detail and the code from the CleanModsForm is horrible to look at in general. The option to select another shader cache cleaning method is never made available.

Please create a new ticket if we want to touch the "cleaning" implementation for a future release.